R/guides-.R

```diff
@@ -566,19 +584,27 @@ Guides <- ggproto(
     stretch_x <- any(unlist(lapply(widths,  unitType)) == "null")
     stretch_y <- any(unlist(lapply(heights, unitType)) == "null")
 
-    # Global justification of the complete legend box
-    global_just <- paste0("legend.justification.", position)
-    global_just <- valid.just(calc_element(global_just, theme))
-
-    if (position == "inside") {
-      # The position of inside legends are set by their justification
-      inside_position <- theme$legend.position.inside %||% global_just
-      global_xjust  <- inside_position[1]
-      global_yjust  <- inside_position[2]
-      global_margin <- margin()
-    } else {
+    if (startsWith(position, "inside")) {
+      # Global justification of the complete legend box
+      global_just <- valid.just(calc_element(
+        "legend.justification.inside", theme
+      ))
+      # for inside guide legends, the position was attached in
+      # each grob of the input grobs (which should share the same position)
+      inside_position <- attr(.subset2(grobs, 1L), "inside_position") %||%
+        # fallback to original method of ggplot2 <=3.5.1
+        .subset2(theme, "legend.position.inside") %||% global_just
       global_xjust  <- global_just[1]
       global_yjust  <- global_just[2]
+      x <- inside_position[1]
+      y <- inside_position[2]
+      global_margin <- margin()
+    } else {
+      # Global justification of the complete legend box
+      global_just <- paste0("legend.justification.", position)
+      global_just <- valid.just(calc_element(global_just, theme))
+      x <- global_xjust  <- global_just[1]
+      y <- global_yjust  <- global_just[2]
```

Comment:
I think we might not even have to alter the logic of `Guides$package_box()` if we just manage everything well in `Guides$assemble()`.

Reply:
Solved

Reply:
I'm happy to explore this if you feel this gets stuck

Reply:
Thanks for your suggestions! After applying them, the code looks much cleaner and more elegant.